extensions/amp-subscriptions/0.1/local-subscription-platform.js

```diff
@@ -25,28 +28,46 @@ export class LocalSubscriptionPlatform {
 
   /**
    * @param {!../../../src/service/ampdoc-impl.AmpDoc} ampdoc
-   * @param {string} serviceUrl
+   * @param {!JsonObject} serviceConfig
    */
-  constructor(ampdoc, serviceUrl) {
+  constructor(ampdoc, serviceConfig) {
     /** @const */
     this.ampdoc_ = ampdoc;
 
-    /** @const @private {string} */
-    this.serviceUrl_ = serviceUrl;
+    /** @const @private {!JsonObject} */
+    this.serviceConfig_ = serviceConfig;
 
     /** @const @private {!../../../src/service/xhr-impl.Xhr} */
     this.xhr_ = Services.xhrFor(this.ampdoc_.win);
   }
 
   /**
-   * TODO(@prateekbh): Define object below once we have a defination of entitlement
-   * @return {!Promise<!JsonObject>}
+   * Returns the Entitlements object for the subscription platform.
+   *
+   * @return {!Promise<!Entitlements>}
    */
   getEntitlements() {
+    user().assert(this.serviceConfig_['paywallUrl'],
+        'service config does not have paywall Url');
+
     return this.xhr_
-        .fetchJson(this.serviceUrl_)
-        .then(res => res.json());
+        .fetchJson(this.serviceConfig_['paywallUrl'])
+        .then(res => res.json())
+        .then(json => {
+          const entitlements = json.entitlements.map(entitlement => {
```

Comment:
Please make sure in the var name it's always clear if this is a JSON or a typed object. E.g. "entitlementJson".